frontend/src/ui-components/AcmTable/AcmTable.tsx

```diff
@@ -787,6 +794,34 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
         })
       }
     }
+
+    const findFilterMatch = (filter: string) => advancedFilters.find((filterItem) => filterItem.id === filter)
+    // advanced filtering, won't apply if preprocessed
+    if (!isPreProcessed) {
+      const advancedFilterSelections: FilterSelections = {}
+      activeAdvancedFilters.forEach(({ columnId, value, operator }) => {
+        if (columnId && value && operator) {
+          advancedFilterSelections[columnId] = [value]
+        }
+      })
+      if (advancedFilters.length && Object.keys(advancedFilterSelections).length) {
+        const filterCategories = Object.keys(advancedFilterSelections)
+
+        filteredItems = filteredItems.filter((item: T) => {
+          let isFilterMatch = true
+          filterCategories.every((filter: string) => {
+            const filterItem: ITableAdvancedFilter<T> | undefined = findFilterMatch(filter)
+            /* istanbul ignore next */
+            const isMatch = filterItem?.tableFilterFn(advancedFilterSelections[filter], item) ?? true
+            if (!isMatch) {
+              isFilterMatch = false
+            }
+            return isMatch
+          })
+          return isFilterMatch
+        })
```

Comment:
Better, but we can still avoid having to use `let` (by actually using the return value of `Array.everey()`) and also declaring the type for `filterItem`.

To test, I set `name` and `namespace` to the same value, and confirmed we hit the filtering function for both. Then I changed `name` to something that does not match any cluster and confirmed we never bothered to check `namespace`.

I notice this was copied from the other filter code, so perhaps we can make this efficiency adjustment there as well, but could be as a separate PR.

```suggestion
        filteredItems = filteredItems.filter((item: T) =>
          filterCategories.every((filter: string) => 
            /* istanbul ignore next */
            findFilterMatch(filter)?.tableFilterFn(advancedFilterSelections[filter], item) ?? true
          )
        )
```

Reply:
I see. I will add that note to my remaining work for iteration 2...